From d98d4202de10332e372fc7bd3492f9fcfeeb7fb1 Mon Sep 17 00:00:00 2001
From: nibon7 <nibon7@gmail.com>
Date: Sat, 24 Aug 2024 12:15:07 +0800
Subject: [PATCH] shell-rs: Don't reset password preference to never when
 unlocking document

We can't save the new password when unlocking document, regardless of
whether you choose to remember it for the session or forever.

Fixes: ed9f9d66ceb9 ("shell-rs: port PpsPasswordView to Rust")
---
 shell-rs/src/password_view.rs | 2 --
 1 file changed, 2 deletions(-)

diff --git a/shell-rs/src/password_view.rs b/shell-rs/src/password_view.rs
index 874645c07..c352985f5 100644
--- a/shell-rs/src/password_view.rs
+++ b/shell-rs/src/password_view.rs
@@ -195,8 +195,6 @@ mod imp {
                 .and_downcast::<gio::SimpleAction>()
                 .unwrap();
 
-            preferences_action.set_state(&glib::Variant::from(PASSWORD_PREFERENCE_NEVER));
-
             match response {
                 RESPONSE_UNLOCK => {
                     let password = dialog.password_entry.text();
-- 
2.44.2

